nixnet/database/_frame.py

```diff
@@ -239,28 +243,24 @@ def mux_subframes(self):
         return self._mux_subframes
 
     @property
-    def pdu_refs(self):
-        return _props.get_frame_pdu_refs(self._handle)
-
-    @pdu_refs.setter
-    def pdu_refs(self, value):
-        _props.set_frame_pdu_refs(self._handle, value)
-
-    @property
-    def pdu_start_bits(self):
-        return _props.get_frame_pdu_start_bits(self._handle)
-
-    @pdu_start_bits.setter
-    def pdu_start_bits(self, value):
-        _props.set_frame_pdu_start_bits(self._handle, value)
-
-    @property
-    def pdu_update_bits(self):
-        return _props.get_frame_pdu_update_bits(self._handle)
-
-    @pdu_update_bits.setter
-    def pdu_update_bits(self, value):
-        _props.set_frame_pdu_update_bits(self._handle, value)
+    def pdus(self):
+        handles = [_pdu.Pdu(handle)
+                   for handle in _props.get_frame_pdu_refs(self._handle)]
+        pdu_tuples = zip(*(handles,
+                           _props.get_frame_pdu_start_bits(self._handle),
+                           _props.get_frame_pdu_update_bits(self._handle)))
+        for pdu in [types.PduFrameReference(ref, start_bit, update_bit)
+                    for (ref, start_bit, update_bit) in pdu_tuples]:
+            yield pdu
+
+    @pdus.setter
+    def pdus(self, pdus):
```

Comment:
#237 changes all ref methods from `object_ref(s)` to `object(s)` and returns the object instead of a ref. Thus, #237 has changed `pdu_refs` to `pdus`. Given that pattern applies to all the other database objects, calling your tuple `pdus` implies the method should return an iterable of PDU objects. Should we perhaps find another name then? Something like `pdu_properties` or `pdu_info`?

Also, you should return a PDU object instead of a handle as well. Just do something like `Pdu(p.ref._handle)`

#237 adds annotations and docstrings for all methods. Moving forward, these will be required for all methods, so you should add these things for your changes as well.